src/main/java/games/strategy/triplea/settings/folders/FolderSettings.java

```diff
@@ -1,51 +1,97 @@
+public class FolderSettings {
 
-  private static final File DEFAULT_DOWNLOADED_MAPS_PATH =
-      new File(ClientFileSystemHelper.getUserRootFolder(), "downloadedMaps");
-  private static final File DEFAULT_SAVE_PATH = new File(ClientFileSystemHelper.getUserRootFolder(), "savedGames");
+  private static final String DEFAULT_MAP_FOLDER = "downloadedMaps";
+  private static final String DEFAULT_SAVE_FOLDER = "savedGames";
 
 
-  @Override
-  public void setToDefault() {
-    setSaveGamePath(DEFAULT_SAVE_PATH.toString());
-    setDownloadedMapPath(DEFAULT_DOWNLOADED_MAPS_PATH.toString());
-  }
+  private final File downloadedMapsFolder;
+  private final File saveGameFolder;
 
-  public String getDownloadedMapPath() {
-    if (validPathFromSystemProperty(SystemPreferenceKey.MAP_FOLDER_OVERRIDE)) {
-      return SystemPreferences.get(SystemPreferenceKey.MAP_FOLDER_OVERRIDE, "");
-    } else if (validPathFromSystemProperty(SystemPreferenceKey.USER_MAPS_FOLDER_PATH)) {
-      return SystemPreferences.get(SystemPreferenceKey.USER_MAPS_FOLDER_PATH, "");
-    } else {
-      return DEFAULT_DOWNLOADED_MAPS_PATH.toString();
-    }
+  public FolderSettings(final PropertyReader propertyFileReader) {
+    this(propertyFileReader, File::mkdirs, DEFAULT_MAP_FOLDER, DEFAULT_SAVE_FOLDER);
   }
 
-  private static boolean validPathFromSystemProperty(SystemPreferenceKey systemProperty) {
-    String value = SystemPreferences.get(systemProperty, "");
-    return !value.isEmpty() && new File(value).exists();
+  @VisibleForTesting
+  FolderSettings(
+      final PropertyReader propertyReader,
+      final Function<File, Boolean> fileMaker,
+      final String defaultMapFolder,
+      final String defaultSaveFolder) {
+    downloadedMapsFolder =
+        ensureFolderExistsOrDie(propertyReader, GameEngineProperty.MAP_FOLDER, defaultMapFolder, fileMaker);
+    saveGameFolder =
+        ensureFolderExistsOrDie(propertyReader, GameEngineProperty.SAVED_GAMES_FOLDER, defaultSaveFolder, fileMaker);
   }
 
+  private static File ensureFolderExistsOrDie(
+      final PropertyReader propertyReader,
+      final GameEngineProperty target,
+      final String defaultValue,
+      final Function<File, Boolean> fileMaker) {
+
+    final String folderPath = propertyReader.readProperty(target, defaultValue);
+    File newFolder = createFolderOrUseDefault(folderPath, defaultValue);
+
+    if (newFolder.exists()) {
+      Preconditions.checkState(newFolder.isDirectory(), newFolder.getAbsolutePath() + " should be a directory");
+      return newFolder;
+    }
+
+    if (!fileMaker.apply(newFolder)) {
+      // failed to create, try the default
+      newFolder = new File(ClientFileSystemHelper.getUserRootFolder(), defaultValue);
+      ClientLogger.logError(
+          "Failed to create folder: " + newFolder.getAbsolutePath()
+              + ", falling back to default: " + newFolder.getAbsolutePath());
```

Comment:
You are logging `newFolder` twice, but my impression is that the first instance should refer to the folder that could not be created when applying `fileMaker`.  Might need a second local variable here so you can capture both folders in the log message.